### PR TITLE
[FEATURE] CombatStone Random

### PR DESCRIPTION
**Pull request by George-Farrell**

### CODE REVIEW

File: src/main/java/org/parabot/randoms/pkhonor/CombatStone [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            Time.sleep(new SleepCondition() {
+                @Override
+                public boolean isValid() {
+                    return combat.distanceTo() > 0 || !combat.getInteractingCharacter().equals(Players.getMyPlayer());
+                }
+            }, 1500);
+        }
+    }
+
+    private Npc getLady() {
```
Comment (SandroCoutinho) [inline src/main/java/org/parabot/randoms/pkhonor/CombatStone]: Rename this method to `getStone()`

File: src/main/java/org/parabot/randoms/pkhonor/CombatStone [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        for (Npc lady : Npcs.getNearest(id)) {
```
Comment (SandroCoutinho) [inline src/main/java/org/parabot/randoms/pkhonor/CombatStone]: Rename field to `stone`

**Review by SandroCoutinho — CHANGES_REQUESTED** [review]

File: src/main/java/org/parabot/randoms/pkhonor/CombatStone [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ */
+public class CombatStone implements Random {
+
+    private Npc combat;
+    private final int id = 17025;
+
+    @Override
+    public boolean activate() {
+        this.combat = getLady();
+        return this.combat != null;
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/CombatStone]:
```
return (this.combat = getLady()) != null;
```
Then you can remove the `this.combat = getLady();` line.

File: src/main/java/org/parabot/randoms/pkhonor/CombatStone [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Override
+    public void execute() {
+        System.out.println("Running CombatStone");
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/CombatStone]: No, either use the logger or use the Parabot output.

**Review by JKetelaar — CHANGES_REQUESTED** [review]

File: src/main/java/org/parabot/randoms/pkhonor/CombatStone [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import org.parabot.environment.api.utils.Time;
+import org.parabot.environment.randoms.Random;
+import org.parabot.environment.randoms.RandomType;
+import org.parabot.environment.scripts.framework.SleepCondition;
+import org.rev317.min.api.methods.Npcs;
+import org.rev317.min.api.methods.Players;
+import org.rev317.min.api.wrappers.Npc;
+
+/**
+ * @author Updated Lord
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/CombatStone]: Is your name `Updated Lord` or just `Lord`? 😉

File: src/main/java/org/parabot/randoms/pkhonor/CombatStone [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+ */
+public class CombatStone implements Random {
+
+    private Npc combat;
+    private final int id = 17025;
+
+    @Override
+    public boolean activate() {
+        this.combat = getCombat();
+        return this.combat != null;
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/CombatStone]:
```
return (this.combat = getCombat()) != null;
```
Then you can remove the `this.combat = getCombat();` line.

**Review by JKetelaar — APPROVED** [review]

**Pull request merged**